CppSamples/EditData/EditGeodatabaseWithTransactions/EditGeodatabaseWithTransactions.cpp

```diff
@@ -0,0 +1,562 @@
+  {
+    featureTypeNames.append(featureType.name());
+  }
+
+  featureTypeNames.sort(Qt::CaseInsensitive);
+  m_currentFeatureTypes = featureTypeNames;
+
+  emit currentFeatureTypesChanged();
+}
+
+void EditGeodatabaseWithTransactions::setSelectedTableName(const QString& tableName)
+{
+  if (m_selectedTableName != tableName)
+  {
+    m_selectedTableName = tableName;
+    emit selectedTableNameChanged();
+
+    // Update feature types for the newly selected table
+    GeodatabaseFeatureTable* table = m_tablesByName.value(tableName, nullptr);
+    if (table)
+    {
+      updateFeatureTypesForTable_(table);
+    }
+  }
+}
+
+void EditGeodatabaseWithTransactions::addFeatureAtLocation(const QString& tableName, const QString& featureTypeName, const QPoint& location)
+{
+  if (!m_mapView || !m_geodatabase)
+  {
+    return;
+  }
+
+  GeodatabaseFeatureTable* targetTable = m_tablesByName.value(tableName, nullptr);
+  if (!targetTable)
+  {
+    setMessageText(QString("Table '%1' not found.").arg(tableName));
+    return;
+  }
+
+  // Find the feature type
+  FeatureType selectedFeatureType;
+  bool foundType = false;
+  for (const FeatureType& featureType : targetTable->featureTypes())
+  {
+    if (featureType.name() == featureTypeName)
+    {
+      selectedFeatureType = featureType;
+      foundType = true;
+      break;
+    }
+  }
+
+  if (!foundType)
+  {
+    setMessageText(QString("Feature type '%1' not found in table '%2'.").arg(featureTypeName, tableName));
+    return;
+  }
+
+  // Convert screen point to map point
+  const Point mapPoint = m_mapView->screenToLocation(location.x(), location.y());
+  if (mapPoint.isEmpty())
+  {
+    setMessageText("Invalid location clicked.");
+    return;
+  }
+
+  // Convert point to WGS84 and check if point is in the extent (extent is in WGS84)
+  const Geometry projectedGeometry = GeometryEngine::project(mapPoint, SpatialReference::wgs84());
+  bool withinBounds = GeometryEngine::contains(m_extent, projectedGeometry);
+  if (!withinBounds)
+  {
+    setMessageText("Error: Feature geometry is outside the generate geodatabase geometry.");
+    return;
+  }
+
+  // Create feature using the selected type's template attributes
+  QVariantMap attributes;
+  if (!selectedFeatureType.templates().isEmpty())
+  {
+    // Use the first template's attributes if available
+    attributes = selectedFeatureType.templates().at(0).prototypeAttributes();
+  }
+
+  Feature* newFeature = targetTable->createFeature(attributes, mapPoint, this);
+  if (!newFeature)
+  {
+    setMessageText("Failed to create new feature.");
+    return;
+  }
+
+  // Add the new feature to the table
+  targetTable->addFeatureAsync(newFeature)
+    .then(this,
+          [this]()
+  {
+    setMessageText("Added feature.");
+  })
+    .onFailed(this,
+              [this](const Error& error)
+  {
+    setMessageText("Error adding feature.");
+    qDebug() << "Add feature failed:" << error.message();
+  })
```

Comment:
All our onFailed handlers will pass our custom exception type. It's fairly concerning this compiles, but that's the compiler's fault and not yours.


```suggestion
              [this](const ErrorException& exception)
  {
    setMessageText("Error adding feature.");
    qDebug() << "Add feature failed:" << exception.error().message();
  })
```